### ISSUE TITLE
Plugins are not installed as logstash user

### ISSUE THREAD

**Issue by alexjfisher** [elastic/puppet-logstash#354]
The [resource defaults](https://github.com/elastic/puppet-logstash/blob/master/manifests/plugin.pp#L120-L125) for the execs don't work as they aren't evaluated before the execs are.

My system umask is 0027, so it's really important that the plugins are installed as the logstash user.

I'm using puppet 5.3.3.  Here is a simple manifest that shows resource defaults don't do anything if they appear after the resources they're supposed to affect.
```puppet
file { '/tmp/testfile':
  ensure => file,
}

File {
  owner => 'logstash',
}
```
```
[root@host ~]# puppet apply test.pp
Notice: Compiled catalog for host.example.com in environment production in 0.02 seconds
Notice: /Stage[main]/Main/File[/tmp/testfile]/ensure: created
Notice: Applied catalog in 0.20 seconds
[root@host ~]# ls -l /tmp/testfile
-rw-r--r--. 1 root root 0 Dec  5 11:47 /tmp/testfile
```

**Comment by jarpy** [comment]
Thanks for spotting this. Resource defaults quietly became parse-order dependent in Puppet 5. I can fix this today.

**Comment by alexjfisher** [comment]
@jarpy Thanks!  I wondered if it was puppet 5 related, as it must have worked at some point.

**Comment by alexjfisher** [comment]
I also see what you mean about the defaults becoming **quietly** parse-order dependent.  The only mention of resource defaults in the any of the 5.x release notes was in relation to https://tickets.puppetlabs.com/browse/PUP-25.  I'm guessing https://github.com/puppetlabs/puppet/pull/5820 has caused the change in behaviour, but it might not have been 100% intentional.

**Comment by jarpy** [comment]
Yeah, I didn't want to seem rude or ungrateful (love you Puppet!), but I had to go on IRC to learn this important fact. Couldn't find a shred of documentation.

**Comment by alexjfisher** [comment]
Oops.  I've just brought it up on slack!

**Comment by jarpy** [comment]
OK. Should be fixed in [6.0.2](https://forge.puppet.com/elastic/logstash/6.0.2).
